app/vminsert/relabel/relabel.go

```diff
@@ -93,10 +93,12 @@ func (ctx *Ctx) ApplyRelabeling(labels []prompb.Label) []prompb.Label {
 	}
 	// Convert src to prompbmarshal.Label format suitable for relabeling.
 	tmpLabels := ctx.tmpLabels[:0]
+	oriNameEmpty := false
 	for _, label := range labels {
 		name := bytesutil.ToUnsafeString(label.Name)
 		if len(name) == 0 {
 			name = "__name__"
+			oriNameEmpty = true
```

Comment:
Every valid metric has a name. It is encoded as an empty byte slice. So the `len(name) == 0` condition is always triggered for valid metric `labels`. This means that `oriNameEmpty` is always set to `true` after this loop (provided that the metrics has a name)

Reply:
So if `labels` contains something like the following:

```go
{
  Name: "__name__",
  Value: "foo"
}
```
then it will be eventually converted to the following after the relabeling:
```go
{
  Name: "",
  Value: "foo",
}
```

This looks like a correct behavior, since `__name__` label is equivalent to a metric name, while metric name is encoded as a label with an empty name inside VictoriaMetrics codebase.

It would be great it you could add a testcase for this issue.

Reply:
after start the relabel function the```__name__``` label is equivalent to a metric name，but GetStorageIdx return different storageNodeIdx between `{
  Name: "__name__",
  Value: "foo"
}` and `{
  Name: "",
  Value: "foo",
}` As a result, storage index writes increase, resulting in service write backlog or even crash